### PR TITLE
Define server api specification

### PR DESCRIPTION
**Pull request by Burbaka**
@roman-mazur @Vandalko please review

### CODE REVIEW

File: src/main/java/com/stanfy/helium/server/spec/public.api [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  data 'Data' optional
+}
+
+service {
+
+  name 'HelliumServer'
+
+  version '0.1'
+  location 'http://helium-1222.appspot.com/'
+  //location 'http://localhost:8080'
```
Comment (Burbaka) [inline src/main/java/com/stanfy/helium/server/spec/public.api:22]: address of GoogleEngine endpoint and for local development

File: build.gradle [hunk 2]
```diff
@@ -17,4 +17,53 @@ targetCompatibility = JavaVersion.VERSION_1_8
...
+  compile 'com.squareup.retrofit:retrofit:1.9.0'
+}
+
+
+import com.stanfy.helium.handler.codegen.java.constants.ConstantNameConverter
+
+import javax.lang.model.element.Modifier
+
+helium {
+  specification file('src/main/java/com/stanfy/helium/server/spec/public.api')
```
Comment (roman-mazur) [inline build.gradle:32]: It should not be in java folder. Something like `src/api` would be better.

File: build.gradle [hunk 3]
```diff
@@ -17,4 +17,53 @@ targetCompatibility = JavaVersion.VERSION_1_8
...
+
+  sourceGen {
+    def sourceOut = file("src/main/api")
```
Comment (roman-mazur) [inline build.gradle:35]: should be `src/generated`

File: .gitignore [hunk 4]
```diff
@@ -68,3 +67,9 @@ gradle-app.setting
 # Linux trash folder which might appear on any partition or disk
 .Trash-*
 
+### Mac ###
+.DS_Store
+
+
+### Hellium spec ###
+/src/main/api
```
Comment (roman-mazur) [inline .gitignore:75]: Not a spec should be ignored but generated source. /src/generated

File: src/main/java/com/stanfy/helium/server/spec/public.api [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+  //////////////////////////////////////
+  // API DECLARATION END
+
+
+  // TESTS FOR API
+  // Each behaviour begins with 'describe'
+  /////////////////////////////////////
+  describe "Could retrieve/delete stored data." spec {
+
+      beforeEach {
```
Comment (roman-mazur) [inline src/main/java/com/stanfy/helium/server/spec/public.api:58]: We use 2 spaces indent.

**Comment by roman-mazur** [comment]
A few comments that you can address in a follow up. Merging.

File: .gitignore (hunk 4 above)
Comment (Burbaka) [inline .gitignore:75]: Actually it was path to gen sources

**Pull request merged**